### ISSUE TITLE
Sample projects have real Excel↔XML drift, now visible since the verify gate works

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#1012]
Split out of #1010, which fixed the build-idempotency gate. The gate had never compared anything, so drift accumulated unseen. With it working, three samples that used to report clean now fail.

| project | before #1010 | after |
|---|---|---|
| TestProject | PASS | **PASS** |
| Poker | PASS | fail |
| SinusitisTherapy | PASS | fail |
| StateTax | PASS | fail |
| CHIP, ChipApp, CorporateTax, Cribbage, KidAid, SyntaxTests, TaxReturn | fail | fail |

TestProject still passing is the evidence that this is real drift and not a false positive from the fix.

## A worked example

StateTax’s committed XML holds

```xml
<condition_dsl>taxpayer.filing_status == "SINGLE"</condition_dsl>
```

while its workbook holds the value **unquoted**:

```
taxpayer.filing_status == SINGLE
```

Excel is the system of record, so by the contract the XML is what is wrong — but note the two are not obviously equivalent: unquoted `SINGLE` is an identifier, not a string literal, so this may be a semantic difference rather than cosmetic. Whichever side is corrected, it should be decided per case rather than by bulk-rebuilding from Excel.

## Why it is not folded into #1010

#1010 is the mechanism; this is content. Rebuilding every sample from its workbook would resolve the diff and could silently change rule behaviour — the `"SINGLE"` case shows why. Each project wants looking at.

`make check` is green: nothing in it verifies these samples, so this is not blocking. It does mean `dtrules verify` cannot yet be run repo-wide as a gate.

## Suggested order

1. StateTax, Poker, SinusitisTherapy — newly failing, so the drift is likely small and recent.
2. The seven already failing, which have other findings (missing Excel, external refs) mixed in.

**Comment by PaulSnow** [comment]
Split out the Cribbage row as #1026 — its failure is not drift: there has never been a workbook, and it also has no map file or scenarios, so it cannot run standalone at all. Different root cause from the StateTax-style Excel↔XML divergence this issue is about.

**Comment by PaulSnow** [comment]
## Refined after attempting the sweep

I said this had become "a mechanical sweep — one `build --from-excel` per project." That is true for two of the six and false for the rest, and I found out by trying it and breaking the build.

**Safe (rebuild changes no DSL and no postfix):**

| project | `[build]` findings | DSL changed |
|---|---|---|
| Poker | 2 → 0 | 0 |
| SinusitisTherapy | 1 → 0 | 0 |

**Not safe:** CHIP, ChipApp, KidAid and TaxReturn rebuild into rules that do not run — filed as #1047. On CHIP, 8 of 65 postfix elements change: two action rows lose their postfix entirely despite their DSL being present in the workbook and compiling standalone, and a local-variable slot shifts from `0 local@` to `1 local@`, which fails at execution with `[OutOfBounds] GetFrameValue`. Five tests that pass against the committed XML fail against the rebuild. The build reports `drops: none` and exits 0.

**StateTax** is the interesting middle case. Its rebuild changes 10 DSL lines — `== "SINGLE"` becomes `== SINGLE` — and I can confirm those compile to identical postfix (`taxpayer.filing_status SINGLE streq`), so it is safe on the evidence. I have not applied it, because doing so alone is not worth a commit while the others are blocked.

## Correction worth recording

My first check for "did the rebuild change behaviour" grepped postfix lines ending in a handful of operators. It reported zero changes for all four projects. It was matching almost nothing — the real comparison, extracting every `*_postfix` element, found the 8 differences immediately. The narrow probe is what let me apply the sweep and break `make check`; reverted.

## Where that leaves this issue

The remaining findings are not one kind of problem:

| kind | meaning | projects |
|---|---|---|
| `[build]` | round-trip drift | all but TestProject |
| `[source]` | missing/invalid `<source>` headers | CHIP, ChipApp, KidAid, CorporateTax, Cribbage, SyntaxTests, TaxReturn |
| `[excel]` | no workbook at all | CorporateTax, Cribbage, SyntaxTests |
| `[external]` | undefined tables/fields/operators | CorporateTax, SyntaxTests, TaxReturn |

Only the first is this issue. The others were always separate and should not be swept up in it.

**Blocked on #1047** for four of the six projects.

**Comment by PaulSnow** [comment]
Status after this session: **8 of 11 samples verify clean**, up from 1.

| sample | status |
|---|---|
| CHIP, ChipApp, KidAid, StateTax | verified — #1071 |
| Poker, SinusitisTherapy | verified — #1059 |
| Cribbage | verified — #1073, also closed #1026 |
| TestProject | was already clean |
| CorporateTax | workbooks bootstrapped; fails on **rule content** — `has_nexus` and `financial` undefined in the NV tables |
| SyntaxTests | bootstraps, but the round trip breaks it — #1074 |
| TaxReturn | reconciled in #1064; full verify takes over ten minutes so it was not re-swept here |

What made this possible was not sample editing — it was five tool defects that
made the drift uncorrectable:

- **#1057** — `build --from-excel` selected the Excel path and then let mtime
  detection answer `NoSync`, so it imported nothing.
- **#1061** — the export guard said "import Excel first"; importing never
  updated the manifest, so the advice could not be followed.
- **#1063** — `RefreshExcelIn` re-created every workbook the manifest named,
  including 114 deleted ones, on every authoring write.
- **#1069** — `sync export`/`import` let detection overrule the direction the
  command names, and `--force` waived the wrong check. This is what blocked the
  XML → Excel push the four behind-workbook samples needed.
- **#1072** — `sync export` refused to create the `excel/` directory it was
  asked to write into, so a project with no Excel could never bootstrap one.

Also corrected along the way: the "14 dropped EDD fields" I reported on #1058
were `Self Reference` and `mapping*key`, both synthesized by
`entity.NewEntity`. Nothing was lost.

Remaining for this issue is two samples with genuine content problems rather
than sync problems, so it may be worth splitting: CorporateTax's undefined
operators want a rules fix, and SyntaxTests is tracked as #1074.

**Comment by PaulSnow** [comment]
Merged as #1114. Ten of eleven samples now verify with **zero** findings.

Three defects, one of them in the checker rather than the rules:

- **A case-sensitive comparison in a case-insensitive language.** The call-graph pass keyed defined tables in a case-sensitive map, so `perform Error_handling_table` did not match `Error_Handling_Table` *defined in the same file*. The struct's own comment claimed the runtime lookup was case-sensitive; it is not — `RName` interns folded. That accounted for all of SyntaxTests' findings and four of TaxReturn's. References now resolve to the authored spelling, so the call edge joins up rather than creating a second node for the same table.
- **Bare words where string literals belong.** NV compared a string field against `financial`, `mining`, `retail` and four more, unquoted, which the compiler read as operator names — the field's own comment lists those as its values. CorporateTax now verifies with nothing at all.
- **Capital-gains rates declared nowhere.** Three tables read `cg_bracket_1_rate` … `cg_bracket_3_rate` and `cg_bracket_N_<filing>`. The thresholds beside them carry the 2025 figures exactly (MFJ $96,700/$600,050, single $48,350/$533,400, HOH $63,000/$551,350), so the tables mean the 0/15/20% long-term rates against those brackets. The rates are declared as data like the thresholds.

`verify` also treated `TEMPLATE_dt.xml` as rules and reported it for naming a workbook that does not exist — which is what a template is for. It now uses the loader's skip predicate.

| | before | after |
|---|---|---|
| samples verifying clean | 8 | **10 of 11** |
| CorporateTax | 7 | 0 |
| SyntaxTests | 2 | 0 |
| TaxReturn | 43 | 23 |

CorporateTax and SyntaxTests are off the CI known-bad list.

**What is left, and why it is different.** TaxReturn's remaining 23 are not mechanical. Fourteen are names like `ftc_limitation`, `taxable_ratio`, `conversion_amount` and `eligible_contributions` standing in for intermediates nobody declared — `set result.foreign_tax_credit = the minimum of taxes_paid and ftc_limitation` is pseudo-code, and completing it means authoring the IRC 904 limitation. Nine are undeclared fields on per-item entities (`property_sale.gain`, `ira_account.type`, `household_employee.wages`), which need both an EDD declaration and a mapping decision about where the data comes from.

That is rules authoring with tax judgment in it, not file repair, so I stopped rather than guess.

**Comment by PaulSnow** [comment]
Closed by #1115. **All eleven samples verify with no findings**, and the CI exclusion list is deleted rather than emptied — a sample failing there is now a regression, not a known defect.

The last 23 were TaxReturn's. Most were near-misses against a field that already existed on the same entity (`property_sale.gain` for `total_gain_loss`, `household_employee.wages` for `cash_wages` — which the same table's own condition already used).

Four were calculations left as placeholders, now written from the forms:

| what | source |
|---|---|
| Credit for the elderly/disabled | Schedule R Part III lines 10–20. The constants were already in the EDD and correct; only the line 15 AGI figures were missing. |
| Foreign tax credit limit | Form 1116 Part III (IRC 904): tax before credits × foreign-source taxable ÷ worldwide taxable. `foreign_source_taxable` was already declared and unused. |
| Roth conversion taxable ratio | Form 8606 lines 6–10 — the pro-rata rule aggregates all traditional/SEP/SIMPLE IRAs, so the year-end value is aggregated rather than per account. |
| Saver's credit | Form 8880 — only the first $2,000 ($4,000 MFJ) counts; the rate was applied to the whole contribution. |

Two things worth recording because they would be easy to get wrong again:

- **Schedule R's MFS AGI base is $5,000**, not $6,250. A secondary summary said otherwise; the form settles it.
- **`fica_rate` was one name doing two jobs.** Schedule H owes both halves of FICA at 15.3% (Pub 926); Form 4137 is the employee share alone at 7.65%. A single constant would have been wrong for one of them whichever value it held. They are two constants now.

And a bug found while checking those rates rather than looked for: `household_ss_threshold` was **2,700 — the 2024 figure** — while every other amount in the project is 2025. Pub 926 puts 2025 at 2,800.
